gluon/scheduler.py

```diff
@@ -978,14 +979,17 @@ def wrapped_pop_task(self):
         contention and retries `pop_task` after 0.5 seconds
         """
         db = self.db
-        db.commit()  # for MySQL only; FIXME: Niphlod, still needed? could avoid when not MySQL?
         for x in range(10):
             try:
+                db.get_connection_from_pool_or_new()
```

Comment:
It's new, probably useful for py4web since it doesn't run the models on every request, it basically calls db._adapter.reconnect. The reason I'm calling it inside the loop is because you could be looping because you lost the connection so it should try to reconnect every time.